internal/installercache/installercache.go

```diff
@@ -41,32 +46,68 @@ func (fi *fileInfo) Compare(other *fileInfo) bool {
 	return fi.info.ModTime().Unix() < other.info.ModTime().Unix()
 }
 
+const (
+	metricEventInstallerCacheRelease = "installercache.release.metrics"
+)
+
 type Release struct {
-	Path string
+	Path          string
+	eventsHandler eventsapi.Handler
+	// enableInstallerCacheEvent should be set to `true` to send metrics upon release, otherwise `false`.
+	enableInstallerCacheEvent bool
+	// startTime is the time at which the request was made to fetch the release.
+	startTime time.Time
+	// clusterID is the UUID of the cluster for which the release is being fetched.
+	clusterID strfmt.UUID
+	// releaseId is the release that is being fetched, for example "4.10.67-x86_64".
+	releaseID string
+	// cached is `true` if the release was found in the cache, otherwise `false`.
+	cached bool
+	// extractDuration is the amount of time taken to perform extraction, zero if no extraction took place.
+	extractDuration float64
 }
 
-func (rl *Release) Release() {
+// Cleanup is called to signal that the caller has finished using the relase and that resources may be released.
+func (rl *Release) Cleanup(ctx context.Context) {
 	if err := os.Remove(rl.Path); err != nil {
 		logrus.New().WithError(err).Errorf("Failed to delete release link %s", rl.Path)
 	}
+	rl.eventsHandler.V2AddMetricsEvent(
```

Comment:
Why should we hide sending a message in the Cleanup method?

Reply:
I don't see this as hiding, I see this as the release sending stats on cleanup.